packages/code-editor/src/Markdown/utils.ts

```diff
@@ -0,0 +1,138 @@
+  } catch (e) {
+    markedContent = content
+  }
+  return markedContent
+}
+
+/**
+ * Removes <mark><mark/> tags from markdown text. The mark tag is removed from code styled
+ * text as the code renderer cannot differentiate <mark><mark/> from the code text.
+ * @param markedText - the markdown text input that contains <mark /> tags
+ */
+const removeMarkTags = (markedText: string) => {
+  return markedText.replace(/<mark>/g, '').replace(/<\/mark>/g, '')
+}
+
+/**
+ * Checks the input for code block decorators, the programming language used (```json), and returns a language tag
+ * that is later used to inform syntax highlighting rules
+ * @param content - the code blob to render
+ * @returns - code blob string with code language tag
+ */
+const addCodeLanguageTags = (content: string) => {
+  let languageTaggedContent
+  try {
+    const replacement = () => '```\n<json/>'
+    const searchPattern = '```json'
+    languageTaggedContent = regReplaceAll(content, searchPattern, replacement)
+  } catch (e) {
+    languageTaggedContent = content
+  }
+  return languageTaggedContent
+}
+
+/**
+ * Removes the code language tag from the code blob text before text presentation to screen
+ * @param content - code blob text with language tag
+ * @returns - code blob text without language tag
+ */
+const removeCodeLanguageTags = (content: string) => {
+  let untaggedContent
+  try {
+    const replacement = () => ''
+    const searchPattern = '<json/>'
+    untaggedContent = regReplaceAll(content, searchPattern, replacement)
+  } catch (e) {
+    untaggedContent = content
+  }
+  return untaggedContent
+}
+
+/**
+ * Extracts the syntax highlighting language, if specified
+ * @param content - language tagged code blob
+ * @returns
+ */
+const getCodeLanguageFromTaggedText = (content: string): string => {
+  return content.match(/<json\/>/g) ? 'json' : 'markup'
+}
+
+/**
+ * Removes tags that were applied for syntax highlighting or search pattern matching and returns just the code blob text
+ * @param content - code blob text to render
+ * @returns - rendered code text
+ */
+export const prepareCodeText = (content: string) => {
+  let text = content
+  const language = getCodeLanguageFromTaggedText(text)
+  text = removeCodeLanguageTags(text)
+  text = removeMarkTags(text)
+  text = text.trim()
+  return { text, language }
+}
+
+/**
+ * Returns a 'qualified markdown' text, a string which contains search pattern match and syntax highlighting qualifiers used by this package
+ * @param content The content to qualify
+ * @param pattern The regex pattern to search
+ * @returns qualified content
+ */
+export const qualifyMarkdownText = (
+  content: string,
+  pattern: string
+): string => {
+  let qualifiedContent
+  if (pattern !== '') {
+    qualifiedContent = addMarkTags(content, pattern)
+  }
+  qualifiedContent = addCodeLanguageTags(qualifiedContent || content)
+  return qualifiedContent
+}
```

Comment:
ReactMarkdown takes the output of this func as input. 

We add `<mark></mark>` tags to all search matches. 

Then we change code block decorators 
```
\`\`\`json
```
to
```
\`\`\`
<json/>
```
So that the code block handler (CodeDisplay) can check for language tag and use it for syntax highlighting rules and remove it before rendering

Reply:
I'm not sure we need to go this route, but let's discuss.